source/distributed/tria.cc

```diff
@@ -3775,6 +3776,187 @@ namespace parallel
+                                                &p4est_cell)
+                        == false))
+        // this quadrant and none of its children belong to us.
+        return;
+
+      // recurse further if both p4est and dealii still have children,
+      // so that we do not run into a dead end in one of the trees
+      bool p4est_has_children = (idx == -1);
+      if ( p4est_has_children && dealii_cell->has_children() )
+        {
+          //recurse further
+          typename dealii::internal::p4est::types<dim>::quadrant
+          p4est_child[GeometryInfo<dim>::max_children_per_cell];
+
+          for (unsigned int c=0; c<GeometryInfo<dim>::max_children_per_cell; ++c)
+            switch (dim)
+              {
+              case 2:
+                P4EST_QUADRANT_INIT(&p4est_child[c]);
+                break;
+              case 3:
+                P8EST_QUADRANT_INIT(&p4est_child[c]);
+                break;
+              default:
+                Assert (false, ExcNotImplemented());
+              }
+
+          dealii::internal::p4est::functions<dim>::
+          quadrant_childrenv (&p4est_cell, p4est_child);
+
+          for (unsigned int c=0;
+               c<GeometryInfo<dim>::max_children_per_cell; ++c)
+            {
+              store_quadrant_cell_relations_recursively (tree,
+                                                         dealii_cell->child(c),
+                                                         p4est_child[c]);
+            }
+        }
+      else if (!p4est_has_children && !dealii_cell->has_children())
+        {
+          // this active cell didn't change
+          // save tuple into corresponding position
+          store_single_quadrant_cell_relation(tree,
+                                              idx,
+                                              dealii_cell,
+                                              Triangulation<dim,spacedim>::CELL_PERSIST);
+        }
+      else if (p4est_has_children) // && !dealii_cell->has_children()
+        {
+          // this cell got refined in p4est, but the dealii_cell has not yet been
+          // refined
+
+          // this quadrant is not active
+          // generate its children, and store information in those
+          typename dealii::internal::p4est::types<dim>::quadrant
+          p4est_child[GeometryInfo<dim>::max_children_per_cell];
+          for (unsigned int c=0; c<GeometryInfo<dim>::max_children_per_cell; ++c)
+            switch (dim)
+              {
+              case 2:
+                P4EST_QUADRANT_INIT(&p4est_child[c]);
+                break;
+              case 3:
+                P8EST_QUADRANT_INIT(&p4est_child[c]);
+                break;
+              default:
+                Assert (false, ExcNotImplemented());
+              }
+
+          dealii::internal::p4est::functions<dim>::
+          quadrant_childrenv (&p4est_cell, p4est_child);
```

Comment:
here and elsewhere in the function, the term `quadrant_childrenv` suggests that it was named before we used a more or less uniform naming scheme anywhere. I don't remember what the `v` was supposed to mean, but maybe just rename this kind of variable to `child_quadrant`.